### PR TITLE
fix: Don't try to upload assets without a tag name

### PR DESCRIPTION
**Pull request by ianlewis**
Updates #1702 

Signed-off-by: Ian Lewis <ianlewis@google.com>

### CODE REVIEW

**Review by asraa — APPROVED** [review]

File: .github/workflows/builder_go_slsa3.yml [hunk 1]
```diff
@@ -329,7 +329,7 @@ jobs:
       contents: write # Needed to write artifacts to a release.
     runs-on: ubuntu-latest
     needs: [build-dry, build, provenance, detect-env]
-    if: inputs.upload-assets == true
+    if: startsWith(github.ref, 'refs/tags/') && inputs.upload-assets
```
Comment (laurentsimon) [inline .github/workflows/builder_go_slsa3.yml:332]: is this correct? I think we want users to be able to upload assets on non-tag triggers, like workflow_dispatch, so long as they can a tag-name to indicate which tag they us to upload to?
This allows some teams to create a release draft out-of-band and give us the tag.

Comment (asraa) [reply .github/workflows/builder_go_slsa3.yml:332]: Ah, nice catch - agreed with the above, either tag or non-empty tag name. then the individual steps below switch on either case

Comment (ianlewis) [reply .github/workflows/builder_go_slsa3.yml]: Yes, you're right I missed this. Updated.

**Comment by ianlewis** [comment]
I'll go ahead and merge this, write e2e tests and point folks to them on the bug.

**Pull request merged**